frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/BaseRebaseBranchOntoParentAction.java

```diff
@@ -106,22 +112,34 @@ protected void doRebase(AnActionEvent anActionEvent, BaseGitMacheteNonRootBranch
     doRebase(project, macheteRepository, gitRepository, branchToRebase);
   }
 
-  private void doRebase(Project project, IGitMacheteRepository macheteRepository, GitRepository repository,
+  private void doRebase(Project project, IGitMacheteRepository macheteRepository, GitRepository gitRepository,
       BaseGitMacheteNonRootBranch branchToRebase) {
+    LOG.debug(() -> "Enter BaseRebaseBranchOntoParentAction#doRebase(project = ${project}, " +
+        "macheteRepository = ${macheteRepository}, gitRepository = ${gitRepository}, " +
+        "branchToRebase = ${branchToRebase} (${branchToRebase.getName()})");
     Try.of(() -> macheteRepository.getParametersForRebaseOntoParent(branchToRebase))
-        .onSuccess(gitRebaseParameters -> new Task.Backgroundable(project, "Rebasing") {
-          @Override
-          public void run(ProgressIndicator indicator) {
-            GitRebaseParams params = getIdeaRebaseParamsOf(repository, gitRebaseParameters);
-            GitRebaseUtils.rebase(project, List.of(repository), params, indicator);
-          }
-
-          // TODO (#95): on success, refresh only sync statuses (not the whole repository). Keep in mind potential
-          // changes to commits (eg. commits may get squashed so the graph structure changes).
-        }.queue()).onFailure(e -> {
+        .onSuccess(gitRebaseParameters -> {
+          LOG.debug(() -> "Queuing \"${branchToRebase.getName()}\" branch rebase background task");
+          new Task.Backgroundable(project, "Rebasing") {
+            @Override
+            public void run(ProgressIndicator indicator) {
+              GitRebaseParams params = getIdeaRebaseParamsOf(gitRepository, gitRebaseParameters);
+              LOG.info(() -> "Rebasing \"${gitRebaseParameters.getCurrentBranch().getName()}\" branch " +
+                  "until ${gitRebaseParameters.getForkPointCommit().getHash()} commit " +
+                  "onto ${gitRebaseParameters.getNewBaseCommit().getHash()}");
+              GitRebaseUtils.rebase(project, List.of(gitRepository), params, indicator);
+            }
+
+            // TODO (#95): on success, refresh only sync statuses (not the whole repository). Keep in mind potential
+            // changes to commits (eg. commits may get squashed so the graph structure changes).
+          }.queue();
+        }).onFailure(e -> {
           // TODO (#172): redirect the user to the manual fork-point
           var message = e.getMessage() == null ? "Unable to get rebase parameters." : e.getMessage();
+          LOG.error(message);
           VcsNotifier.getInstance(project).notifyError("Rebase failed", message);
+          GuiUtils.invokeLaterIfNeeded(() -> Messages.showErrorDialog(message, "Something Went Wrong..."),
```

Comment:
Nit: `Something went wrong...`

Reply:
Capitalizing first letters is JB convention here

Reply:
While we are here I suggest handling the `selectedBranch` presence like in `RebaseSelectedBranchOntoParentAction`. So do it in `update` method. In the case of the absence make action invisible. 

Actually there is an interesting case here cause we do not check whether the selected branch belongs to the git machete repository. Is it a bug or a feature? :smile: If it is a bug then add this check to update too. 

If this PR is gonna be merged after #217 then `com.virtuslab.gitmachete.frontend.actions.ActionUtils#getSelectedMacheteBranch` might be useful.

Reply:
@mkondratek @micpiotrowski is there an issue for that/does it fall under any of the existing issues? Pls open one if not

Reply:
Make it here https://github.com/VirtusLab/git-machete-intellij-plugin/issues/220